egui/src/widgets/plot/mod.rs

```diff
@@ -694,88 +723,31 @@ impl PreparedPlot {
             return;
         }
 
-        let interact_radius: f32 = 16.0;
-        let mut closest_value = None;
-        let mut closest_item = None;
-        let mut closest_dist_sq = interact_radius.powi(2);
-        for item in items {
-            if let Some(values) = item.values() {
-                for value in &values.values {
-                    let pos = transform.position_from_value(value);
-                    let dist_sq = pointer.distance_sq(pos);
-                    if dist_sq < closest_dist_sq {
-                        closest_dist_sq = dist_sq;
-                        closest_value = Some(value);
-                        closest_item = Some(item.name());
-                    }
-                }
-            }
-        }
-
-        let mut prefix = String::new();
-        if let Some(name) = closest_item {
-            if !name.is_empty() {
-                prefix = format!("{}\n", name);
-            }
-        }
-
-        let line_color = if ui.visuals().dark_mode {
-            Color32::from_gray(100).additive()
-        } else {
-            Color32::from_black_alpha(180)
-        };
+        let interact_radius_sq: f32 = (16.0f32).powi(2);
 
-        let value = if let Some(value) = closest_value {
-            let position = transform.position_from_value(value);
-            shapes.push(Shape::circle_filled(position, 3.0, line_color));
-            *value
-        } else {
-            transform.value_from_position(pointer)
-        };
-        let pointer = transform.position_from_value(&value);
+        let candidates = items.iter().filter_map(|item| {
+            let item = &**item;
+            let closest = item.find_closest(pointer, transform);
 
-        let rect = transform.frame();
+            Some(item).zip(closest)
+        });
 
-        if *show_x {
-            // vertical line
-            shapes.push(Shape::line_segment(
-                [pos2(pointer.x, rect.top()), pos2(pointer.x, rect.bottom())],
-                (1.0, line_color),
-            ));
-        }
-        if *show_y {
-            // horizontal line
-            shapes.push(Shape::line_segment(
-                [pos2(rect.left(), pointer.y), pos2(rect.right(), pointer.y)],
-                (1.0, line_color),
-            ));
-        }
+        let closest = candidates
+            .min_by_key(|(_, elem)| elem.dist_sq.ord())
+            .filter(|(_, elem)| elem.dist_sq <= interact_radius_sq);
```

Comment:
I don't think that `float_ord.rs` is really required. I'd just replace all `min_by_key` with `min_by`, for example here:

` .min_by(|(_, elem1), (_, elem2)| elem1.dist_sq.partial_cmp(&elem2.dist_sq).unwrap())`

Reply:
I was doing that first, but it's _much_ less readable and easier to get wrong, no?
```rs
.min_by(|(_, elem1), (_, elem2)| elem1.dist_sq.partial_cmp(&elem2.dist_sq).unwrap())
```
vs.
```rs
.min_by_key(|(_, elem)| elem.dist_sq.ord())
```
Also, we would need to do that for any sort/min/max related algorithm, so I was thinking it makes sense to abstract it once and reuse it? The idea would be that maybe other parts of egui could benefit as well (now or in the future).

Reply:
I agree with @Bromeon here, especially if we make the `OrderedFloat` also handle `NaN`. Then it is strictly better (avoids panics).